mimic/model/maas_objects.py

```diff
@@ -0,0 +1,229 @@
+        data = self._get_default_data()
+
+        if override_key in self._overrides:
+            data_fn = self._overrides[override_key]
+            data = data_fn(timestamp)
+
+        return {'type': self.type,
+                'unit': self._unit,
+                'data': data}
+
+
+class CheckType(object):
+    """
+    Data model for a MaaS check type (e.g., remote.ping).
+    """
+    def __init__(self, clock, metrics):
+        """
+        Constructs a check type with metrics.
+        """
+        self.metrics = metrics
+        self.test_check_available = {}
+        self.test_check_status = {}
+        self.test_check_response_code = {}
+        self._clock = clock
+
+    def clear_overrides(self):
+        """
+        Clears the overrides for test-checks and metrics.
+        """
+        self.test_check_available = {}
+        self.test_check_status = {}
+        self.test_check_response_code = {}
+
+        for metric in self.metrics:
+            metric.clear_overrides()
+
+    def get_metric_by_name(self, metric_name):
+        """
+        Gets the metric on this check type.
+
+        This method is useful for setting and clearing overrides on the
+        test metrics.
+        """
+        for metric in self.metrics:
+            if metric.name == metric_name:
+                return metric
+        raise NameError('No metric named "{0}"!'.format(metric_name))
+
+    def get_test_check_response(self, **kwargs):
+        """
+        Gets the response as would have been returned by the test-check API.
+        """
+        entity_id = kwargs['entity_id']
+        check_id = kwargs.get('check_id', '__test_check')
+        monitoring_zones = kwargs.get('monitoring_zones') or ['__AGENT__']
+
+        ench_key = (entity_id, check_id)
+        timestamp = int(1000 * self._clock.seconds())
+
+        return (self.test_check_response_code.get(ench_key, 200),
+                [{'timestamp': timestamp,
+                  'monitoring_zone_id': monitoring_zone,
+                  'available': self.test_check_available.get(ench_key, True),
+                  'status': self.test_check_status.get(
+                      ench_key, 'code=200,rt=0.4s,bytes=99'),
+                  'metrics': dict([(m.name,
+                                    m.get_value_for_test_check(
+                                        entity_id=entity_id,
+                                        check_id=check_id,
+                                        monitoring_zone=monitoring_zone,
+                                        timestamp=timestamp))
+                                   for m in self.metrics])}
+                 for monitoring_zone in monitoring_zones])
+
+
+class MaasStore(object):
+    """
+    A collection of MaaS configuration objects.
+    """
+    def __init__(self, clock):
+        """
+        Initializes the Maas configuration using the provided clock.
+        """
+        self.check_types = {
+            'agent.cpu': CheckType(clock, [
```

Comment:
Could you provide a documentation citation in a docstring to indicate where this blob of values comes from?